packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -2087,6 +2088,9 @@ function commitRootImpl(
     if (!rootDoesHavePassiveEffects) {
       rootDoesHavePassiveEffects = true;
       pendingPassiveEffectsRemainingLanes = remainingLanes;
+      // workInProgressTransitions might be overwritten, so we want
+      // to store it in pendingPassiveTransitions until they get processed
+      pendingPassiveTransitions = workInProgressTransitions;
```

Comment:
This needs to be passed in as an argument to `commitRoot` similar to `recoverableErrors`

Reply:
> Because workInProgressTransitions might change before we process it in the passive effects, we introduce a new variable, pendingPassiveTransitions

This is true but it's also true that it might change in between the commit phase and the render phase, because we sometimes throttle the commit with `setTimeout`.